app/controllers/tasks.js

```diff
@@ -262,16 +276,62 @@ export default class TasksController extends Controller {
       );
     }
   }
-
-  @action async handleUpdateTask(taskId, error) {
+  showTaskChangeInfoModal(msg) {
+    this.message = msg;
+    this.showModal = true;
+    this.buttonRequired = true;
+  }
+  @action async handleUpdateTask(taskId, resetCurrentTask) {
+    this.resetCurrentTask = resetCurrentTask;
     const taskData = this.taskFields;
-    if (taskData.percentCompleted === TASK_PERCENTAGE.completedPercentage) {
-      this.message = TASK_MESSAGES.MARK_DONE;
-      this.showModal = true;
-      this.buttonRequired = true;
-      this.tempTaskId = taskId;
+    const currentTask = this.getTaskById(taskId);
+    const isCurrentTaskStatusBlock =
+      currentTask.status === this.TASK_KEYS.BLOCKED;
+    const isCurrentTaskStatusInProgress =
+      this.TASK_KEYS.IN_PROGRESS === currentTask.status;
+    const isNewTaskStatusInProgress =
+      taskData.status === this.TASK_KEYS.IN_PROGRESS;
+    const isNewTaskStatusBlock = taskData.status === this.TASK_KEYS.BLOCKED;
+    const isCurrProgress100 =
+      parseInt(currentTask.percentCompleted || 0) === 100;
+    const isCurrProgress0 = parseInt(currentTask.percentCompleted || 0) === 0;
```

Comment:
the logic for determining whether to proceed with the task update or show a modal dialogue can be refactored for better readability and maintainability. Consider breaking down the logic into smaller, more focused functions.